app/models/slide.rb

```diff
@@ -68,8 +68,13 @@ def page_list_url
   end
 
   def page_list
-    url = self.page_list_url
-    get_json(url)
+    len = num_of_pages.abs.to_s.length
+    result = []
+    for i in 1..num_of_pages
+      n = i.to_s.rjust(len, '0')
```

Comment:
__[reek]__
* [UncommunicativeVariableName] has the variable name 'n' [:ref](https://github.com/troessner/reek/blob/master/docs/Uncommunicative-Variable-Name.md)